src/_pytest/assertion/rewrite.py

```diff
@@ -427,7 +427,22 @@ def _saferepr(obj: object) -> str:
     sequences, especially '\n{' and '\n}' are likely to be present in
     JSON reprs.
     """
-    return saferepr(obj).replace("\n", "\\n")
+    maxsize = _get_maxsize_for_saferepr(util._config)
```

Comment:
This consults the Config at the time the test is run. Alternative is to consult the config at the time the test is *rewritten*, that is, make the `AssertionRewriter` get the maxsize, add a `maxsize` argument to `_saferepr` and make `AssertionRewriter` pass it as a constant in the AST. Do you have any preference? The reason I'm talking about that is trying to avoid the `util._config` mutable global. These tend to be not great...

Reply:
> Alternative is to consult the config at the time the test is rewritten, that is, make the AssertionRewriter get the maxsize

I thought that too, however I realized assertion rewritten is cached, so if you invoke `pytest` the first time, the `.pyc` file is rewritten and cached; if you invoke `pytest -v` next, the `.pyc` file is cached so assertion rewriting is not triggered. Unfortunately this information needs to be available at runtime.

I injected the `Config` object in the same way we inject the comparison functions today, so I think this at least aligns with how things work today, however I'm definitely open for alternative ideas. 👍